uvtools/plot.py

```diff
@@ -723,3 +723,185 @@ def plot_diff_uv(uvd1, uvd2, pol=None, check_metadata=True, bins=50):
+        Default behavior is to use the 'both' setting.
+
+    check_metadata : bool, optional
+        Whether to check that the metadata for `uvd1` and `uvd2` match.
+        See ``utils.check_uvd_pair_metadata`` docstring for details on 
+        how the metadata are compared. If `check_metadata` is set to 
+        False, but the metadata don't agree, then the plotter may or 
+        may not error out, depending on how the metadata disagree. 
+        Default behavior is to check the metadata.
+    
+    dimension : str, optional
+        String specifying which dimension is used for the base domain. This 
+        may be either 'time' or 'freq'. Default is to determine which axis has
+        more entries and to use that axis.
+
+    Returns
+    -------
+    fig : matplotlib.pyplot.Figure
+        Figure object containing the plots. The plots have their axes and 
+        titles automatically set depending on what quantities are being 
+        plotted.
+
+    Notes
+    -----
+    This function extracts the visibility waterfall corresponding to the 
+    provided antpairpol and flattens it by taking the mean along the axis 
+    not being used. For true 1d arrays this just modifies the shape, but it 
+    might not exactly be the desired behavior--so be warned.
+    """
+    if check_metadata:
+        utils.check_uvd_pair_metadata(uvd1, uvd2)
+
+    if plot_type not in ("base", "dual", "both"):
+        raise ValueError(
+            "You must specify whether to make one or two plots with "
+            "the ``plot_type`` parameter. You may choose to plot the "
+            "visibility difference as a function of frequency/time by "
+            "setting ``plot_type`` to 'base', or you can choose to "
+            "plot the difference in Fourier space by setting "
+            "``plot_type`` to 'dual'. If you would like to plot both, "
+            "then set ``plot_type`` to 'both'."
+        )
+
+    dimensions_to_duals = {"time" : "fr", "freq" : "dly"} 
+
+    if dimension is None:
+        dimension = "time" if uvd1.Ntimes > uvd1.Nfreqs else "freq"
+        if uvd1.Ntimes == uvd1.Nfreqs:
+            warnings.warn(
+                "The UVData objects passed have the same number of " \
+                "times as they do frequencies. You did not specify " \
+                "which dimension to use, so the difference plots " \
+                "will be made along the time axis."
+            )
+
+    if dimension not in ("time", "freq"):
+        raise ValueError(
+            "You must specify whether the visibilities are a function "
+            "of time or frequency by setting the ``dimension`` "
+            "parameter to 'time' or 'freq', respectively."
+        )
+    
+    dual = dimensions_to_duals[dimension]
+
+    use_axis = 0 if dimension == "time" else 1
+    proj_axis = (use_axis + 1) % 2
+
+    # get visibility data
+    vis1 = uvd1.get_data(antpairpol).mean(axis=proj_axis)
```

Comment:
I can imagine sometimes wanting to plot the median instead.

Reply:
I've updated the code to allow the user to specify which `numpy` averaging function they would like to use, with the default being `np.mean`. This also allows the user to do a weighted average using `np.average`, so I've added `**kwargs` to the function signature, and these are passed directly to the averaging function used.